shards/shards_test.go

```diff
@@ -284,6 +287,111 @@ func TestUnloadIndex(t *testing.T) {
 	}
 }
 
+func TestShardedSearcher_List(t *testing.T) {
+	repos := []*zoekt.Repository{
+		{
+			ID:        1234,
+			Name:      "repo-a",
+			Branches:  []zoekt.RepositoryBranch{{Name: "main"}, {Name: "dev"}},
+			RawConfig: map[string]string{"repoid": "1234"},
+		},
+		{
+			Name:     "repo-b",
+			Branches: []zoekt.RepositoryBranch{{Name: "main"}, {Name: "dev"}},
+		},
+	}
+
+	// Test duplicate removal when ListOptions.Minimal is true and false
+	ss := newShardedSearcher(4)
+	ss.replace("1", searcherForTest(t, testIndexBuilder(t, repos[0])))
+	ss.replace("2", searcherForTest(t, testIndexBuilder(t, repos[0])))
+	ss.replace("3", searcherForTest(t, testIndexBuilder(t, repos[1])))
+	ss.replace("4", searcherForTest(t, testIndexBuilder(t, repos[1])))
+
+	for _, tc := range []struct {
+		name string
+		opts *zoekt.ListOptions
+		want *zoekt.RepoList
+	}{
+		{
+			name: "nil opts",
+			opts: nil,
+			want: &zoekt.RepoList{
+				Repos: []*zoekt.RepoListEntry{
+					{
+						Repository: *repos[0],
+						Stats:      zoekt.RepoStats{Shards: 2},
+					},
+					{
+						Repository: *repos[1],
+						Stats:      zoekt.RepoStats{Shards: 2},
+					},
+				},
+			},
+		},
+		{
+			name: "minimal=false",
+			opts: &zoekt.ListOptions{Minimal: false},
+			want: &zoekt.RepoList{
+				Repos: []*zoekt.RepoListEntry{
+					{
+						Repository: *repos[0],
+						Stats:      zoekt.RepoStats{Shards: 2},
+					},
+					{
+						Repository: *repos[1],
+						Stats:      zoekt.RepoStats{Shards: 2},
+					},
+				},
+			},
+		},
+		{
+			name: "minimal=true",
+			opts: &zoekt.ListOptions{Minimal: true},
+			want: &zoekt.RepoList{
+				Repos: []*zoekt.RepoListEntry{
+					{
+						Repository: *repos[1],
+						Stats:      zoekt.RepoStats{Shards: 2},
+					},
+				},
+				Minimal: map[uint32]*zoekt.MinimalRepoListEntry{
+					repos[0].ID: {
+						HasSymbols: repos[0].HasSymbols,
+						Branches:   repos[0].Branches,
+					},
+				},
+			},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			q := &query.Repo{Pattern: "epo"}
+
+			res, err := ss.List(context.Background(), q, tc.opts)
+			if err != nil {
+				t.Fatalf("List(%v, %s): %v", q, tc.opts, err)
+			}
+
+			sort.Slice(res.Repos, func(i, j int) bool {
+				return res.Repos[i].Repository.Name < res.Repos[j].Repository.Name
+			})
+
+			ignored := []cmp.Option{
+				cmpopts.EquateEmpty(),
+				cmpopts.IgnoreFields(zoekt.RepoListEntry{}, "IndexMetadata"),
+				cmpopts.IgnoreFields(zoekt.RepoStats{}, "IndexBytes", "Documents"),
```

Comment:
@tsenart I merged in https://github.com/sourcegraph/zoekt/pull/98 so you can remove this ignore after rebasing.